provider/provider.go

```diff
@@ -125,12 +125,8 @@ func (p *dockerNativeProvider) Check(ctx context.Context, req *rpc.CheckRequest)
 		return nil, err
 	}
 
-	// Get the system platform
-	os := runtime.GOOS
-	// Docker for Mac builds and runs for linux/amd64, not darwin
-	if os == "darwin" {
-		os = "linux"
-	}
+	// OS defaults to Linux in all cases
+	os := "linux"
 	arch := runtime.GOARCH
 	hostPlatform := os + "/" + arch
```

Comment:
I wonder if we could peek into the `moby/moby` repo or `docker/cli` and see what their logic is. If they infer a platform value, should we import their method?

Reply:
Ah! I did verify that on macOS with an empty `platform` string, the daemon does assume reasonable OS defaults, i.e. linux on macOS.